pkg/scheduler/plugins/reservation/reservation.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reservation
+
+import (
+	"k8s.io/klog"
+	"time"
+
+	"volcano.sh/volcano/pkg/scheduler/api"
+	"volcano.sh/volcano/pkg/scheduler/framework"
+	"volcano.sh/volcano/pkg/scheduler/util"
+)
+
+// PluginName defines the name of the plugin
+const PluginName = "reservation"
+
+type reservationPlugin struct {
+	pluginArguments framework.Arguments
+}
+
+// New returns the plugin instance
+func New(arguments framework.Arguments) framework.Plugin {
+	return &reservationPlugin{pluginArguments: arguments}
+}
+
+func (rp *reservationPlugin) Name() string {
+	return PluginName
+}
+
+func (rp *reservationPlugin) OnSessionOpen(ssn *framework.Session) {
+	// select the job which has the highest priority and waits for the longest duration
+	targetJobFn := func(jobs []*api.JobInfo) *api.JobInfo {
```

Comment:
There is a question worth discussing, why use `targetJob` the whole PR while not something like `bigJob` instead?

Reply:
> There is a question worth discussing, why using `targetJob` the whole PR while not something like `bigJob` instead?

In fact, selected job is at the high priority and waits for a long time. So we consider it as target job. Usually, it is a big job which keeps pending for lack of resource.  But that's not the only case.